pkg/plugins/policies/meshhttproute/api/v1alpha1/meshhttproute.go

```diff
@@ -53,7 +53,44 @@ type RuleConf struct {
 	BackendRefs *[]BackendRef `json:"backendRefs,omitempty"`
 }
 
+// +kubebuilder:validation:Enum=RequestHeaderModifier;ResponseHeaderModifier
+type FilterType string
+
+const (
+	RequestHeaderModifierType  FilterType = "RequestHeaderModifier"
+	ResponseHeaderModifierType FilterType = "ResponseHeaderModifier"
+)
+
+// +kubebuilder:validation:MinLength=1
+// +kubebuilder:validation:MaxLength=256
+// +kubebuilder:validation:Pattern=`^[A-Za-z0-9!#$%&'*+\-.^_\x60|~]+$`
+type HeaderName string
+
+type HeaderKeyValue struct {
+	Name  HeaderName `json:"name"`
+	Value string     `json:"value"`
```

Comment:
Should we add `Append` here? Both MeshHealthCheck and MeshRateLimit have it

Reply:
This is the Gateway API compatible version

Reply:
Okay, then MeshHealthCheck and MeshRateLimit should probably have:
```go
Header HeaderKeyValue `json:"name"`
Append *bool `json:"append,omitempty"`
```
Could you please move `HeaderKeyValue` to `api/common`? I'll update other policies to use it

Reply:
nit: maybe just `HeaderValue` instead of `HeadeKeyValue`

Reply:
I think the idea was to use set/add for other policies too

Reply:
We could move `HeaderModifier` to common, but other policies don't support `remove`, so I'm not sure what we should do about it.

Reply:
Well most important is that the API is similar for users. If we want to share the types we move the one without `remove` to common and have this one embed that and add `remove`

Reply:
I'm resolving this since we have https://github.com/kumahq/kuma/issues/5712